### PR TITLE
Add CI guest instance health checks

### PR DESCRIPTION
**Pull request by wangxian-ibm**
CI keeps on failing FIP check, adding these health checks to ensure guest instance's health before FIP check.

### CODE REVIEW

**Comment by wangxian-ibm** [comment]
Should I not use the term "health check" because "health check" actually refers to our sensu stuff?

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

**Comment by wangxian-ibm** [comment]
retest

File: playbooks/tests/tasks/integration.yml [hunk 1]
```diff
@@ -105,6 +105,43 @@
     volume: turtle-vol
   when: cinder.enabled|bool
 
+- name: instance is active
```
Comment (nirajdp76) [inline playbooks/tests/tasks/integration.yml]: what happens if instance take little longer to go to active state?
probably we should have retry logic, 6 times with 10sec delay between each retry.

File: playbooks/tests/tasks/integration.yml [hunk 2]
```diff
@@ -105,6 +105,43 @@
...
+  shell: . /root/stackrc; openstack server show {{ turtle_stack.openstack.id }} | grep ACTIVE
+
+- name: instance is running
```
Comment (nirajdp76) [inline playbooks/tests/tasks/integration.yml]: probably we should have retry logic, 6 times with 10sec delay between each retry.

File: playbooks/tests/tasks/integration.yml [hunk 3]
```diff
@@ -105,6 +105,43 @@
...
+  shell: . /root/stackrc; openstack server show {{ turtle_stack.openstack.id }} | grep Running
+
+- name: instance ready for login
```
Comment (nirajdp76) [inline playbooks/tests/tasks/integration.yml]: probably we should have retry logic, 6 times with 10sec delay between each retry.

File: playbooks/tests/tasks/integration.yml [hunk 4]
```diff
@@ -105,6 +105,43 @@
...
+
+- name: neutron dhcp  namespace can ping instance
+  shell: DHCP_NS=$( ip netns show | grep qrouter- | awk '{print $1}' );
+         ip netns exec ${DHCP_NS} ping -c 5 {{ turtle_stack.openstack.private_v4 }}
+  register: result
+  until: result|success
+  retries: 6
+  delay: 10
+
+- name: neutron router namespace can ping internet
```
Comment (nirajdp76) [inline playbooks/tests/tasks/integration.yml]: why do we need 6 re-tries here?

**Comment by wangxian-ibm** [comment]
@nirajdp76 I will add retries/wait on active and running checks. We should also increase the wait time for checking if instance is ready for login because metadata and dhcp sometimes takes longer. In this case, we can remove the retries/wait for ping the instance because if the instance is ready for login, it should respond to ping. Otherwise the instance is in bad state and it should just fail.

**Comment by nirajdp76** [comment]
retest

**Comment by nirajdp76** [comment]
retest

**Comment by epwn** [comment]
This looks good to me

File: playbooks/tests/tasks/integration.yml (hunk 1 above)
Comment (wangxian-ibm) [inline playbooks/tests/tasks/integration.yml]: added retries/wait

**Comment by nirajdp76** [comment]
retest

**Pull request merged**